### PR TITLE
Alert Dialogs made uniform in both modes

### PR DESCRIPTION
**Pull request by ritik2017**
Fixes Issue #913 

Changes: The dialog width was set to 95% of screen width and the styles for light theme were modified.

Screenshots/GIF for the change: 
screenshot_2019-01-15-16-44-50-511_org kiwix kiwixcustomexample 1
screenshot_2019-01-15-16-44-35-250_org kiwix kiwixcustomexample 1

### CODE REVIEW

**Comment by soloturn** [comment]
@ritik2017 , thank you so much for the pull request. currently the pull reuqest has 9 commits in it, instead of one. it would be much easier to review if you would take out things not belonging to the topic, like update gradle, remove unused imports. squashing the commit into one, and having a more concise commit message in the lines of : https://gist.github.com/matthewhudson/1475276 also helps.

**Comment by soloturn** [comment]
i know this whole interactive rebasing stuff sunds lenghty. as you work on your own repository, and your own branch you can avoid this interactive rebasing right from the beginning by ammending to your commit:
```
$ git commit -m "same message as before" --amend
$ git push -f
```

**Comment by ritik2017** [comment]
Thanks @soloturn for explaining the stuff.. I am working on squashing the commits into one

**Comment by ritik2017** [comment]
Not sure but i think the issue #927 reports this

**Comment by soloturn** [comment]
@ritik2017 thanks you!! now the code makes a lot more sense now. with selecting "r" for reword you would be able to adjust the commit comments to match this code quality now.

https://github.com/kiwix/kiwix-android/pull/919/commits/cf5da6b6e07030466b38f37ce72c8f9b26fa9920

the message for commit cf5da6b6e is 35 lines long, and it does not mention what you fixed: Make dialog boxes uniform for both light and dark modes. maybe you could remove all other text and just include what is fixed? an example for a commit message is here: https://gist.github.com/matthewhudson/1475276 

https://github.com/kiwix/kiwix-android/pull/919/commits/c88572dc8f44bf99407c2ea1dfa1aae7b582fcb8

the message for commit  c88572dc is just two words "handle exception". i see that you added Objects.requireNonNull, if you could add in the message why this helps, and with what? i googled a little and i found in stackoverflow: https://stackoverflow.com/questions/45632920/why-should-one-use-objects-requirenonnull/45632962 which says that this would make it fail earlier?


https://github.com/kiwix/kiwix-android/pull/919/commits/b6cc0567d9234279465e268bcc9f2b80c352f1fa

the commit b6cc0567d seems to undo some of the above. maybe you could squash all three commits then?

**Comment by ritik2017** [comment]
@soloturn Thanks a lot for going through all the commits so patiently.. I am working to resolve the issue.

**Comment by soloturn** [comment]
thank you so much @ritik2017, excellent, imo this can be put on master!! the commit message is a little short for people with less epertise than yours - but that is not blocking anything. ideally the commit should be readable and understandable without the ticket and the pull request so one can browse through the history of commits without clicking anything else.

File: app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java [hunk 1]
```diff
@@ -924,20 +926,19 @@ public void openExternalUrl(Intent intent) {
...
-        .setNeutralButton(R.string.do_not_ask_anymore, (dialogInterface, i) -> {
-          sharedPreferenceUtil.putPrefExternalLinkPopup(false);
-          isExternalLinkPopup = false;
-
-          startActivity(intent);
-        })
-        .setPositiveButton(android.R.string.yes, (dialogInterface, i) -> startActivity(intent))
-        .setIcon(android.R.drawable.ic_dialog_alert)
-        .show();
+            .setTitle(R.string.external_link_popup_dialog_title)
```
Comment (soloturn) [inline app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java]: when rebasing it onto master, and formatting the method via ctrl-shift-alt-L, i did not get a change here, same as before? you are sure you want to commit this class? to me it looks like the change is in app/src/main/res/values/styles.xml only.

File: app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java [hunk 2]
```diff
@@ -74,6 +74,7 @@
 import android.widget.FrameLayout;
 import android.widget.ImageButton;
 import android.widget.ImageView;
+import android.widget.LinearLayout;
```
Comment (soloturn) [inline app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java]: the code below did not change, are you sure it needs an additional import?

File: app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java [hunk 3]
```diff
@@ -103,6 +104,7 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Locale;
+import java.util.Objects;
```
Comment (soloturn) [inline app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java]: the code below did not change, are you sure it needs an additional import?

**Comment by ritik2017** [comment]
@soloturn sorry forgot to remove the import statement have done that now and pushed the code..
Hope this resolves the issue

**Comment by soloturn** [comment]
@ritik2017 thanks! can you please sqaush the 2 commits so there is one commit with one file changed?

**Comment by ritik2017** [comment]
Thanks a lot @soloturn i hope the issue is resolved now

**Comment by soloturn** [comment]
thanks @ritik2017, looks already really promising, one change and we are there!! please take out the unchanged file MainActivity.java from the commit. google says this could be done the following way:
* https://stackoverflow.com/questions/12481639/remove-files-from-git-commit
* https://help.github.com/articles/removing-files-from-a-repository-s-history/
your situation is slightly different, as you are allowed to do force push.

**Comment by ritik2017** [comment]
@soloturn I hope only the useful stuff of code remains in commit now which also fixes the issue
Thanks a lot for going through the commit so patiently and explaining the errs in detail.. It was really very  helpful going through all the posts and learning new stuffs... Hope to keep these in mind in my next commits and PRs

**Comment by soloturn** [comment]
@ritik2017 thank you! now the commit contains what you changed only. you tested the change and it works? your first comment on this page said:
"Changes: The dialog width was set to 95% of screen width and the styles for light theme were modified."

@abdulwd , @siddharth2010 what is your usual process of testing and rolling out?

**Comment by ritik2017** [comment]
@soloturn i have tested the app and it works fine
I had earlier set the width to 95% but later realised that the issue is fixed without doing that so i removed that peice of code in MainActivity and it still works fine

**Comment by abdulwd** [comment]
@soloturn I test the changes on my device and then they are approved and merged.

**Comment by abdulwd** [comment]
@ritik2017 Nice work! @mhutti1 It is ready to be merged.

**Comment by abdulwd** [comment]
@ritik2017 Sorry, I pressed `Update branch` by mistake. I have pushed the changes again it will soon be merged.

**Comment by ritik2017** [comment]
@abdulwd Thanks for review
@soloturn Thanks a lot for all your help and assistance
Working to resolve other issues

**Comment by siddharth2010** [comment]
@soloturn @ritik2017 @abdulwd 
I just noticed that the 'clear history' one had a different icon in the top left from the 'enter external link' dialog box.

image

I think we should also make all the icons in the dialog boxes uniform as well.

Should we do it in this PR itself, or do you think we should open another ticket for this

**Comment by abdulwd** [comment]
I think we should open another ticket for it.

**Comment by ritik2017** [comment]
Shall i work on the issue??
@siddharth2010 @abdulwd @soloturn

**Comment by siddharth2010** [comment]
@ritik2017 Sure go ahead. I will open another ticket for this and you can work on that.

**Pull request merged**